File: app/src/main/java/com/example/eugenia/game_2048/MusicService.java
package com.example.eugenia.game_2048;

import android.app.Service;
import android.content.Intent;
import android.media.MediaPlayer;
import android.os.IBinder;

public class MusicService extends Service {
    private MediaPlayer musicPlayer;
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }
    @Override
    public void onCreate() {
        super.onCreate();
        musicPlayer = MediaPlayer.create(this,R.raw.adagio);
        musicPlayer.setLooping(true);
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        musicPlayer.start();
        return super.onStartCommand(intent, flags, startId);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        musicPlayer.stop();
    }
}

File: Fahrenheit/app/src/main/java/com/example/fahrenheit/MusicService.kt
package com.example.fahrenheit

import android.app.Service
import android.content.Context
import android.content.Intent
import android.media.MediaPlayer
import android.os.IBinder
import androidx.annotation.Nullable


class MusicService : Service() {

    private var mPlayer: MediaPlayer? = null


    @Nullable
    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        val extra = intent.extras
        val pref = getSharedPreferences("LIGHT_MOD", Context.MODE_PRIVATE)
        val isMusicale = pref.getBoolean("MUSIC", true)
        if (extra != null) {
            val idMusic = extra.getInt("MUSIC")
            mPlayer = MediaPlayer.create(this, idMusic)
            mPlayer!!.isLooping = false
            if (isMusicale)
                mPlayer!!.start()
        }
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onDestroy() {
        super.onDestroy()
        val pref = getSharedPreferences("LIGHT_MOD", Context.MODE_PRIVATE)
        val isMusicable = pref.getBoolean("MUSIC", true)
        if (isMusicable)
            mPlayer!!.stop()
    }
}